### PR TITLE
Update Zoom API authentication from JWT to server-to-server OAuth

### PR DESCRIPTION
**Pull request by elenabaurkot**
# Description
We previously used a Zoom App with JWT authentication for Zoom API requests, but Zoom is deprecating JWT starting 6/1/2023. We created a new Server-to-Server OAuth Zoom App for our Zoom API authentication, which continues to allow for our API calls to be authenticated without the need for user authentication. The new server-to-server Zoom app required adding scopes (scopes define the app permissions for which API methods the app is allowed to call), if new API methods are added to the Zoom API, the scopes will need to be added to the server-to-server Zoom app on Zoom Marketplace as well ([more details on how to do that outlined here](https://github.com/bebraven/platform/wiki/Zoom#adding-new-zoom-api-calls)). I tested using the "Generate Zoom Links" tool and generating/canceling Zoom links via the sync and Zoom API calls were working properly. 

# Motivation and Context
To make sure our Zoom authentication continues to work. 

Task: https://app.asana.com/0/1201131148207877/1202436246726540/f

### CODE REVIEW

File: lib/zoom_api.rb [hunk 1]
```diff
@@ -17,23 +17,23 @@ class TooManyRequestsError < StandardError; end
...
+      :client_secret => Rails.application.secrets.zoom_client_secret
     }
-    token = JWT.encode(payload, Rails.application.secrets.zoom_api_secret, 'HS256')
+    access_token_url = "https://zoom.us/oauth/token?grant_type=account_credentials&account_id=#{Rails.application.secrets.zoom_account_id}"
+    token_response = RestClient.post(access_token_url, auth_params)
+    token_response_json = JSON.parse(token_response.body)
 
-    new(token)
+    access_token = token_response_json['access_token']
+    new(access_token)
```
Comment (tomlovett) [inline lib/zoom_api.rb:29]: Nice work! Very well structured and easy to follow.

**Review by tomlovett — APPROVED** [review]

**Review by rshipp — APPROVED** [review]
did it end up expiring after an hour, or does it refresh itself automatically somehow?

**Comment by elenabaurkot** [comment]
It ended up expiring after the hour, I guess you just have to make a new request for an access token after every hour, but it requests a new access token every time we call `Zoom.client` so it didn't end up being a problem @rshipp

**Comment by sadleb** [comment]
@elenabaurkot makes me happy to see how little code it took to switch this. Not the case in my new job!

**Comment by elenabaurkot** [comment]
> @elenabaurkot makes me happy to see how little code it took to switch this. Not the case in my new job!

Way less than I though it was going to take @sadleb !

**Pull request merged**